sdk/core/core-http/src/policies/exponentialRetryPolicy.ts

```diff
@@ -142,96 +133,47 @@ export class ExponentialRetryPolicy extends BaseRequestPolicy {
   }
 }
 
-/**
- * Determines if the operation should be retried and how long to wait until the next retry.
- *
- * @param {ExponentialRetryPolicy} policy The ExponentialRetryPolicy that this function is being called against.
- * @param {number} statusCode The HTTP status code.
- * @param {RetryData} retryData  The retry data.
- * @return {boolean} True if the operation qualifies for a retry; false otherwise.
- */
-function shouldRetry(
-  policy: ExponentialRetryPolicy,
-  statusCode: number | undefined,
-  retryData: RetryData
-): boolean {
-  if (
-    statusCode == undefined ||
-    (statusCode < 500 && statusCode !== 408) ||
-    statusCode === 501 ||
-    statusCode === 505
-  ) {
-    return false;
-  }
-
-  let currentCount: number;
-  if (!retryData) {
-    throw new Error("retryData for the ExponentialRetryPolicyFilter cannot be null.");
-  } else {
-    currentCount = retryData && retryData.retryCount;
-  }
-
-  return currentCount < policy.retryCount;
-}
-
-/**
- * Updates the retry data for the next attempt.
- *
- * @param {ExponentialRetryPolicy} policy The ExponentialRetryPolicy that this function is being called against.
- * @param {RetryData} retryData  The retry data.
- * @param {RetryError} [err] The operation"s error, if any.
- */
-function updateRetryData(
-  policy: ExponentialRetryPolicy,
-  retryData?: RetryData,
-  err?: RetryError
-): RetryData {
-  if (!retryData) {
-    retryData = {
-      retryCount: 0,
-      retryInterval: 0
-    };
-  }
-
-  if (err) {
-    if (retryData.error) {
-      err.innerError = retryData.error;
-    }
-
-    retryData.error = err;
-  }
-
-  // Adjust retry count
-  retryData.retryCount++;
-
-  // Adjust retry interval
-  let incrementDelta = Math.pow(2, retryData.retryCount) - 1;
-  const boundedRandDelta =
-    policy.retryInterval * 0.8 +
-    Math.floor(Math.random() * (policy.retryInterval * 1.2 - policy.retryInterval * 0.8));
-  incrementDelta *= boundedRandDelta;
-
-  retryData.retryInterval = Math.min(incrementDelta, policy.maxRetryInterval);
-
-  return retryData;
-}
-
-function retry(
+async function retry(
   policy: ExponentialRetryPolicy,
   request: WebResourceLike,
   response?: HttpOperationResponse,
   retryData?: RetryData,
   requestError?: RetryError
 ): Promise<HttpOperationResponse> {
-  retryData = updateRetryData(policy, retryData, requestError);
+  retryData = updateRetryData(
+    {
+      retryInterval: policy.retryInterval,
+      minRetryInterval: 0,
+      maxRetryInterval: policy.maxRetryInterval
+    },
+    retryData,
+    requestError
+  );
+  function initialCheck(response?: HttpOperationResponse, _error?: RetryError) {
```

Comment:
also you can omit the unused param

```suggestion
  function initialCheck(response?: HttpOperationResponse) {
```

should still be valid I think

Reply:
Done